### PR TITLE
using pdll for sparse reshape elimination

### PR DESCRIPTION
**Pull request by Orion34-lanbo**

### CODE REVIEW

File: tao_compiler/mlir/disc/transforms/disc_pdl_utils.cc [hunk 1]
```diff
@@ -145,7 +146,7 @@ OwningOpRef<ModuleOp> compilePDLL(
       op->erase();
     }
   }
-  if (VLOG_IS_ON(1)) {
+  if (1 /*VLOG_IS_ON(1)*/) {
```
Comment (wyzero) [inline tao_compiler/mlir/disc/transforms/disc_pdl_utils.cc:149]: remove debug code？

File: tao_compiler/mlir/disc/transforms/disc_pdl_utils.cc [hunk 2]
```diff
@@ -227,6 +228,24 @@ static LogicalResult checkConstantTensor(PatternRewriter& rewriter,
   return matchPattern(v, m_Constant(&denseAttr)) ? success() : failure();
 }
 
+static LogicalResult checkConstantTensorValueIsOne(PatternRewriter& rewriter,
```
Comment (wyzero) [inline tao_compiler/mlir/disc/transforms/disc_pdl_utils.cc:231]: better to use Attribute as const value input？ e.g. CheckTensorValueIsOne(val, attr<"1">)?

File: tao_compiler/mlir/disc/transforms/disc_pdl_utils.cc (hunk 1 above)
Comment (Orion34-lanbo) [reply tao_compiler/mlir/disc/transforms/disc_pdl_utils.cc]: done

File: tao_compiler/mlir/disc/transforms/disc_pdl_utils.cc (hunk 2 above)
Comment (Orion34-lanbo) [reply tao_compiler/mlir/disc/transforms/disc_pdl_utils.cc]: done

**Review by wyzero — APPROVED** [review]

**Review by qiuxiafei — APPROVED** [review]
LGTM~ Nice implementation with PDLL~

**Comment by Orion34-lanbo** [comment]
to #510

**Pull request merged**